### PR TITLE
fix(upgrad engine) Prompt message when upgrading engine automatically

### PR DESCRIPTION
**Pull request by smallteeths**
https://github.com/longhorn/longhorn/issues/2281

Signed-off-by: Siye Wang <siye@rancher.com>

### CODE REVIEW

File: src/utils/status.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+const isAutomaticallyUpgradeEngine = (volume, engineUpgradePerNodeLimit, defaultEngineImage) => {
+  if (engineUpgradePerNodeLimit && engineUpgradePerNodeLimit.value !== '0') {
+    if (defaultEngineImage) {
+      return volume.engineImage !== defaultEngineImage.image
+    }
+    return false
+  }
+  return false
+}
```
Comment (PhanLe1010) [inline src/utils/status.js:12]: To determine whether a volume is upgrading, we only need to check `volume. engineImage  != volume. currentImage`
We don't need to compare it to the `defaultEngineImage`

```
export function statusUpgradingEngine(volume) {
   if (volume && volume. engineImage && volume. currentImage) {
      return volume. engineImage  != volume. currentImage
   }
   return false
 }
```

File: src/utils/status.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  // - Not DR volume
+  // - Not degraded volume
+  // - Not expanding volume
+  // - In upgrading process (i.g., curretnImage != engineImage )
+  // - concurrent-automatic-engine-upgrade-per-node-limit setting is bigger than 0
+  if (!volume.standby && volume.robustness !== 'degraded' && volume.size === oldSize && isAutomaticallyUpgradeEngine(volume, engineUpgradePerNodeLimit, defaultEngineImage)) {
+    return (<Tooltip title={`Volume engine is being upgraded from ${volume.engineImage} to ${defaultEngineImage.image}`}><Icon style={{ marginRight: 5 }} type="sync" spin /></Tooltip>)
+  }
+  return ''
+}
```
Comment (PhanLe1010) [inline src/utils/status.js:28]: This is the condition to show the global notification. We don't need this to check if an individual volume is upgrading.

**Review by PhanLe1010 — CHANGES_REQUESTED** [review]

File: src/utils/status.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  // - Not DR volume
+  // - Not degraded volume
+  // - Not expanding volume
+  // - In upgrading process (i.g., curretnImage != engineImage )
+  // - concurrent-automatic-engine-upgrade-per-node-limit setting is bigger than 0
+  if (!volume.standby && volume.robustness !== 'degraded' && volume.size === oldSize && isAutomaticallyUpgradeEngine(volume, engineUpgradePerNodeLimit)) {
+    return (<Tooltip title={`Volume engine is being upgraded from ${volume.engineImage} to ${volume.currentImage}`}><Icon style={{ marginRight: 5 }} type="sync" spin /></Tooltip>)
+  }
+  return ''
+}
```
Comment (PhanLe1010) [inline src/utils/status.js:27]:
```
const isUpgradingEngine = (volume) => {
   if (volume && volume.engineImage && volume.currentImage) {
      return volume.engineImage  != volume.currentImage
   }
   return false
 }

export function statusUpgradingEngine(volume) {
   if (isUpgradingEngine(volume)) {
     return (<Tooltip title={`Upgrading engine from ${volume.engineImage} to ${volume.currentImage}`}><Icon style={{ marginRight: 5 }} type="sync" spin /></Tooltip>)
   }
   return ''
 }
```

**Review by PhanLe1010 — CHANGES_REQUESTED** [review]

**Review by PhanLe1010 — APPROVED** [review]
LGTM

**Review by yasker — APPROVED** [review]
Approve on behavior of @PhanLe1010

**Pull request merged**